hack/jenkins/minishift_linux_integration_tests_kvm.sh

```diff
@@ -0,0 +1,232 @@
+  exit_with_message "$?" "Error starting Minishift VM"
+  output=`$BINARY status | sed -n 1p`
+  assert_equal "$output" "Minishift:  Running"
+  print_success_message "Starting VM"
+}
+
+function verify_stop_instance() {
+  $BINARY stop
+  exit_with_message "$?" "Error starting Minishift VM"
+  output=`$BINARY status | sed -n 1p`
+  assert_equal "$output" "Minishift:  Stopped"
+  print_success_message "Stopping VM"
+}
+
+function verify_swap_space() {
+  output=`$BINARY ssh -- echo $(free | tail -n 1 | awk '{print $2}')`
+  if [ "$output" == "0" ]; then
+    echo "Expected '$output' not equal to 0"
+    result=1
+  fi
+  print_success_message "Swap space check"
+}
+
+function verify_ssh_connection() {
+  output=`$BINARY ssh -- echo hello`
+  assert_equal $output "hello"
+  print_success_message "SSH Connection"
+}
+
+function verify_vm_ip() {
+  output=`$BINARY ip`
+  assert_valid_ip $output
+  print_success_message "Getting VM IP"
+}
+
+function verify_cifs_installation() {
+  expected="mount.cifs version: 6.2"
+  output=`$BINARY ssh -- sudo /sbin/mount.cifs -V`
+  assert_equal "$output" "$expected"
+  print_success_message "CIFS installation"
+}
+
+function verify_sshfs_installation() {
+  expected="SSHFS version 2.5"
+  output=`$BINARY ssh -- sudo sshfs -V`
+  echo $output | grep "$expected" > /dev/null
+  if [ "$?" != "0" ]; then
+    echo "Expected exit status of command '$BINARY ssh -- sudo sshfs -V' to be 0."
+    result=1
+  fi
+  print_success_message "SSHFS installation"
+}
+
+function verify_nfs_installation() {
+  expected="mount.nfs: (linux nfs-utils 1.3.0)"
+  output=`$BINARY ssh -- sudo /sbin/mount.nfs -V /need/for/version`
+  assert_equal "$output" "$expected"
+  print_success_message "NFS installation"
+}
+
+function verify_bind_mount() {
+  output=`$BINARY ssh -- 'findmnt | grep "\[/var/lib/" | wc -l'`
+  assert_equal $output "4"
+  print_success_message "Bind mount check"
+}
+
+function verify_hvkvp_installation() {
+  expected="Error opening pool"
+  output=`$BINARY ssh -- 'hvkvp'`
+  assert_equal "$output" "$expected"
+  print_success_message "HVKVP check"
+}
+
+function verify_delete() {
+  $BINARY delete --force
+  exit_with_message "$?" "Error deleting Minishift VM"
+  output=`$BINARY status`
+  if [ "$1" != "$2" ]; then
+    echo "Expected '$1' equal to '$2'"
+    result=1
+  fi
+  print_success_message "Deleting VM"
+}
+
+# Tests
+verify_start_instance
+sleep 90
+verify_swap_space
+verify_vm_ip
+verify_cifs_installation
+verify_sshfs_installation
+verify_nfs_installation
+verify_bind_mount
+verify_delete
+
+if [[ $result -eq 0 ]]; then
+  status="success"
+else
+  status="failure"
+  cat $HOME/.kube/config
+  echo $PATH
+
+  docker ps
+
+  ${BINARY} status
+  ${BINARY} ip
+  ${BINARY} ssh -- cat /etc/VERSION
+  ${BINARY} ssh -- docker ps
+  ${BINARY} openshift logs
+fi
+
+set +x
+target_url="https://storage.googleapis.com/minikube-builds/logs/${MINIKUBE_LOCATION}/${JOB_NAME}.txt"
```

Comment:
The test isn't stuck, its actually running and failing, but the errors above need to be caught or ignored so that this line can get ran, which updates the github status line.